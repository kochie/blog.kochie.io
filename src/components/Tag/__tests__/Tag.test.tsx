import React from 'react'
import { ReactTestRenderer, act, create } from 'react-test-renderer'

import { Tag, TagSet } from '..'

describe('TAG COMPONENT', () => {
  test('should render', () => {
    let tree: ReactTestRenderer

    act(() => {
      tree = create(<Tag name={'tagName'} link={'tagLink'} />)
    })

    // @ts-expect-error tree will be assigned
    expect(tree.toJSON()).toMatchSnapshot()
  })
})

describe('TAGSET COMPONENT', () => {
  test('should render', () => {
    let tree: ReactTestRenderer

    act(() => {
      tree = create(
        <TagSet>
          <Tag name={'tagName'} link={'tagLink'} />
        </TagSet>
      )
    })

    // @ts-expect-error tree will be assigned
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
