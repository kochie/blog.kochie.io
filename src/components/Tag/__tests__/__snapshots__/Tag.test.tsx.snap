// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TAG COMPONENT should render 1`] = `
<a
  href="tagLink"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  <div
    className="tag tag-common"
  >
    tagName
  </div>
</a>
`;

exports[`TAGSET COMPONENT should render 1`] = `
<div
  className="undefined tagSet"
>
  <a
    href="tagLink"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="tag tag-common"
    >
      tagName
    </div>
  </a>
</div>
`;
